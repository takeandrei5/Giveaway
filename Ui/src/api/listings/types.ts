import { PaginatedResult } from '../../utils/types';

export type ItemData = { id: string; image: string; title: string; createdAt: Date };

export type CreateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export type UpdateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export interface FetchListingsResponse {
	listings: PaginatedResult<ItemData>;
}

export type ListingInformation = {
	id: string;
	title: string;
	description: string;
	category: number;
	createdAt: Date;
	images: string[];
};

export type OwnerInformation = {
	email: string;
	name: string;
	image: string;
};

export type FetchListingDetailsResponse = { listingInfo: ListingInformation } & { ownerInfo: OwnerInformation };
