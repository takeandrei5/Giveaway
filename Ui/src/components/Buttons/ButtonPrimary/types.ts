import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = Omit<ButtonProps, 'color'>;
