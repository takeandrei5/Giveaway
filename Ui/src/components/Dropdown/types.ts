export type DropdownProps = {
	id: string;
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
	name: string;
	value: string | number | readonly string[];
	isInvalid?: false;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
