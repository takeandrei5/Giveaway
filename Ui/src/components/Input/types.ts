export type InputProps = {
	id: string;
	name: string;
	placeholder: string;
	value?: string;
	onBlur?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onChange?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	disabled?: boolean;
	multiline?: boolean;
	isInvalid?: boolean;
	height?: string | number;
	label?: string;
	leftIcon?: JSX.Element;
	rightIcon?: JSX.Element;
	rows?: number;
	width?: string | number;
};
