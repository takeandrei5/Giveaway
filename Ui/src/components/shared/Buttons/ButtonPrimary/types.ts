import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = ButtonProps;
