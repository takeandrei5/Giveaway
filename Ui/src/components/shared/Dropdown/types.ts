export type DropdownProps = {
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
