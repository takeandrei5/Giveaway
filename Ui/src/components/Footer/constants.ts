import { StackItem } from './types';

const stackItems: StackItem[] = [
	{
		stackColumnName: 'Product',
		stackColumnItems: ['Overview', 'Features', 'Tutorials', 'Pricing', 'Releases'],
	},
	{
		stackColumnName: 'Company',
		stackColumnItems: ['About', 'Press', 'Careers', 'Contact', 'Partners'],
	},
	{
		stackColumnName: 'Support',
		stackColumnItems: ['Help Center', 'Terms of Service', 'Legal', 'Privacy Policy', 'Status'],
	},
	{
		stackColumnName: 'Follow us',
		stackColumnItems: ['Facebook', 'Twitter', 'Github', 'Instagram', 'Linkedin'],
	},
];

export { stackItems };
