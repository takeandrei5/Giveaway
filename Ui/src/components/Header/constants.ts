const DEFAULT_AVATAR: string = 'https://avatars.dicebear.com/api/male/username.svg';

export { DEFAULT_AVATAR };
