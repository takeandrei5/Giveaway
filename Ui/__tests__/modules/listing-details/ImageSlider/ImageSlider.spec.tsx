import { ImageSlider } from '@modules/listing-details/ImageSlider';
import { ImageSliderProps } from '@modules/listing-details/ImageSlider/types';
import { render } from '@testing-library/react';
import { TEST_IMAGE } from '__tests__/constants';

describe('ImageSlider', () => {
	it('should match snapshot', () => {
		// Arrange
		const props: ImageSliderProps = {
			images: [TEST_IMAGE],
		};

		// Act
		const { container } = render(<ImageSlider {...props} />);

		// Assert
		expect(container).toMatchSnapshot();
	});
});
