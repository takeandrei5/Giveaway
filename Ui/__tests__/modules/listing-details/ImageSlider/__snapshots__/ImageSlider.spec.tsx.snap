// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider should match snapshot 1`] = `
<div>
  <div
    class="css-17vrnyc"
  >
    <div
      class="slick-slider slick-initialized"
    >
      <div
        class="slick-list"
        style="height: 0px;"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="css-kiasv6"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="image"
                    data-nimg="responsive"
                    decoding="async"
                    draggable="false"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="width: 0px; display: block; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    tabindex="-1"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
